File: plus-minus/plus-minus.go
package main

import "fmt"

func plusMinus(arr []int32) {
	positiveCount := 0
	negativeCount := 0
	zeroCount := 0

	for _, n := range arr {
		if n > 0 {
			positiveCount++
		} else if n < 0 {
			negativeCount++
		} else {
			zeroCount++
		}
	}

	length := float32(len(arr))
	fmt.Println(float32(positiveCount) / length)
	fmt.Println(float32(negativeCount) / length)
	fmt.Println(float32(zeroCount) / length)
}

File: plus-minus/PlusMinus.java
import java.util.List;

public class PlusMinus {

  public static void plusMinus(List<Integer> ns) {
    int positiveCount = 0;
    int negativeCount = 0;
    int zeroCount = 0;

    for (int n : ns) {
      if (n > 0) {
        positiveCount++;
      } else if (n < 0) {
        negativeCount++;
      } else {
        zeroCount++;
      }
    }

    List
      .of(positiveCount, negativeCount, zeroCount)
      .stream()
      .map(count -> (double) count / (double) ns.size())
      .forEach(System.out::println);
  }

}
